src/program.cpp

```diff
@@ -205,8 +208,70 @@ target_assignments program::get_target_assignments(const std::vector<target>& ta
 
 bool program::is_compiled() const { return not this->impl->target_name.empty(); }
 
+void program::compile(const std::vector<std::string>& targets,
+                      std::unordered_map<std::string, compile_options> compile_opt_map)
+{
+    // Gather all the target roots
+    std::unordered_multimap<std::string, module_ref> roots;
```

Comment:
I find it better to keep it `<std::string, module_ref>` rather than referencing different targets by index.  Atleast for now `compile()`.  We keep `target` as std::string for each root module so that makes it better.

Reply:
> I find it better to keep it <std::string, module_ref> rather than referencing different targets by index.

But if two targets have the same name then we can't distinguish them.

Reply:
Yes, but right now `make_target()` only construct target based on string target_name input. So if targets are different then, their names must be different at the moment.

Reply:
Made it `map<size_t, module_ref>`.